### PR TITLE
Incorporate compiler plugin code-generators for unique service-path generation

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#2487](https://github.com/ballerina-platform/ballerina-standard-library/issues/2487)

## Examples
N/A

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: ballerina/annotation.bal [hunk 1]
```diff
@@ -44,3 +44,9 @@ public type SubscriberServiceConfiguration record {|
 
 # WebSub Subscriber Configuration for the service, indicating subscription related parameters.
 public annotation SubscriberServiceConfiguration SubscriberServiceConfig on service;
+
+public type MetaInformation record {|
+    byte[] servicePath;
+|};
+
+public annotation MetaInformation MetaInfo on service;
```
Comment (chamil321) [inline ballerina/annotation.bal:52]: Is this public annotation? If so we need docs

File: ballerina/annotation.bal [hunk 2]
```diff
@@ -44,3 +44,9 @@ public type SubscriberServiceConfiguration record {|
 
 # WebSub Subscriber Configuration for the service, indicating subscription related parameters.
 public annotation SubscriberServiceConfiguration SubscriberServiceConfig on service;
+
+public type MetaInformation record {|
```
Comment (chamil321) [inline ballerina/annotation.bal:48]: Need docs

File: ballerina/sub_listener.bal [hunk 3]
```diff
@@ -111,9 +106,18 @@ public class Listener {
         self.externAttach(completeSevicePath, 'service, httpService, serviceConfig);
     }
 
-    isolated function retrieveGeneratedServicePath(SubscriberService subscriberService) returns string|Error = @java:Method {
-        'class: "io.ballerina.stdlib.websub.NativeWebSubListenerAdaptor"
-    } external;
+    isolated function retrieveGeneratedServicePath(SubscriberService subscriberService) returns string|Error {
+        MetaInformation? metaInfo = retrieveServiceMetaInfo(subscriberService);
+        if (metaInfo is MetaInformation) {
```
Comment (chamil321) [inline ballerina/sub_listener.bal:111]:
```suggestion
        if metaInfo is MetaInformation {
```

File: ballerina/sub_listener.bal [hunk 4]
```diff
@@ -111,9 +106,18 @@ public class Listener {
...
+            string|error servicePath = strings:fromBytes(metaInfo.servicePath);
+            if servicePath is error {
+                return error Error("Error retrieving Service Information", servicePath);
```
Comment (chamil321) [inline ballerina/sub_listener.bal:114]:
```suggestion
                return error Error("Error retrieving service information", servicePath);
```

File: ballerina/sub_listener.bal [hunk 5]
```diff
@@ -111,9 +106,18 @@ public class Listener {
...
+            }
+            return servicePath;
+        } else {
+            return error Error("Error retrieving Service Information: service path not found");
```
Comment (chamil321) [inline ballerina/sub_listener.bal:118]: unnecessary capital letters

File: compiler-plugin/src/main/java/io/ballerina/stdlib/websub/task/ServiceMetaInfoUpdatorTask.java [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+import java.util.Map;
+import java.util.Optional;
+
+import static io.ballerina.stdlib.websub.task.service.path.ServicePathContextHandler.getContextHandler;
+
+/**
+ * {@code serviceMetaInfoUpdatorTask} modifies the source by adding required meta-info for the websub service
+ * declarations.
+ */
+public class ServiceMetaInfoUpdatorTask implements ModifierTask<SourceModifierContext> {
```
Comment (chamil321) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/websub/task/ServiceMetaInfoUpdatorTask.java:66]:
```suggestion
public class ServiceMetaInfoUpdaterTask implements ModifierTask<SourceModifierContext> {
```

File: ballerina/annotation.bal (hunk 1 above)
Comment (ayeshLK) [reply ballerina/annotation.bal]: Fixed with [bdfe5bb4dc40458438ae4849d58d8f61bb6840e3](https://github.com/ballerina-platform/module-ballerina-websub/pull/510/commits/bdfe5bb4dc40458438ae4849d58d8f61bb6840e3)

File: ballerina/annotation.bal (hunk 2 above)
Comment (ayeshLK) [reply ballerina/annotation.bal]: Fixed with [bdfe5bb4dc40458438ae4849d58d8f61bb6840e3](https://github.com/ballerina-platform/module-ballerina-websub/pull/510/commits/bdfe5bb4dc40458438ae4849d58d8f61bb6840e3)

File: ballerina/sub_listener.bal (hunk 3 above)
Comment (ayeshLK) [reply ballerina/sub_listener.bal]: Fixed with [bdfe5bb4dc40458438ae4849d58d8f61bb6840e3](https://github.com/ballerina-platform/module-ballerina-websub/pull/510/commits/bdfe5bb4dc40458438ae4849d58d8f61bb6840e3)

File: ballerina/sub_listener.bal (hunk 4 above)
Comment (ayeshLK) [reply ballerina/sub_listener.bal]: Fixed with [bdfe5bb4dc40458438ae4849d58d8f61bb6840e3](https://github.com/ballerina-platform/module-ballerina-websub/pull/510/commits/bdfe5bb4dc40458438ae4849d58d8f61bb6840e3)

File: ballerina/sub_listener.bal (hunk 5 above)
Comment (ayeshLK) [reply ballerina/sub_listener.bal]: Fixed with [bdfe5bb4dc40458438ae4849d58d8f61bb6840e3](https://github.com/ballerina-platform/module-ballerina-websub/pull/510/commits/bdfe5bb4dc40458438ae4849d58d8f61bb6840e3)

File: compiler-plugin/src/main/java/io/ballerina/stdlib/websub/task/ServiceMetaInfoUpdatorTask.java (hunk 6 above)
Comment (ayeshLK) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/websub/task/ServiceMetaInfoUpdatorTask.java]: Fixed with [56fc9380bb7384ae76c9a996c6b795c91b5cc982](https://github.com/ballerina-platform/module-ballerina-websub/pull/510/commits/56fc9380bb7384ae76c9a996c6b795c91b5cc982)

**Review by chamil321 — APPROVED** [review]

**Pull request merged**